services/QuillLMS/config/initializers/secure_headers.rb

```diff
@@ -1,5 +1,24 @@
+csp_types = %w(default_src script_src font_src img_src style_src connect_src)
+permissive_config = csp_types.each_with_object({}) do |n, memo|
```

Comment:
There are some shorter ways to write this current code using enumerable methods, but it might just be clearer (and possibly shorter overall) to just explicitly write out the full config, e.g.
```ruby
permissive_config = {
  default_src: ["*"],
  font_src: ["*"],
  script_src: [
      "*", 
      "'unsafe-inline'",
      "'unsafe-eval'",
  ],
  ...
```
  I would have found this easier to parse and compare with the `default_config`

Reply:
Yes - this is not the best syntax for long term code. However, it is helpful when I need to iterate rapidly over the config - sometimes applying changes to all keys, and sometimes applying changes to individual keys. I plan to remove all of `permissive_config` in a matter of days when it becomes obsolete.